preload/modulepreload.html

```diff
@@ -59,6 +59,36 @@
     });
 }, 'link rel=modulepreload');
 
+promise_test(function(t) {
```

Comment:
Should this be a utility function that accepts the different URLs and `as` values?

Reply:
Latest commit revamps this to take an array of module types for shared tests. That way we can have unique tests for each module type. This reduced a lot of duplication. It's not used everywhere because a lot of these tests are unique for script, but it'll make it much easier to expand for future module types like "text".